src/pkg/bundle/deploy_test.go

```diff
@@ -616,3 +616,197 @@ func TestFileVariableHandlers(t *testing.T) {
+	)
+
+	type TestCase struct {
+		name        string
+		bundle      Bundle
+		loadEnv     bool
+		expectedKey string
+		expectedVal string
+		envKey      string
+		envVal      string
+	}
+
+	setUpPkg := func(overVar types.BundleChartVariable) types.Package {
+		return types.Package{Name: pkgName,
+			Overrides: map[string]map[string]types.BundleChartOverrides{componentName: {chartName: {Variables: []types.BundleChartVariable{
+				overVar,
+			}}}}}
+	}
+
+	testCases := []TestCase{
+		{
+			name: "simple path, set by config",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{
+						Config: "uds-config.yaml",
+						Variables: map[string]map[string]interface{}{
+							pkgName: {
+								"VAR1": "set-by-config",
+							},
+						},
+					},
+				},
+				bundle: types.UDSBundle{
+					Packages: []types.Package{setUpPkg(types.BundleChartVariable{Name: "VAR1", Path: "path"})},
+				},
+			},
+			expectedKey: "VAR1",
+			expectedVal: "set-by-config",
+		},
+		{
+			name: "complex path, set by bundle",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{},
+				},
+				bundle: types.UDSBundle{
+					Packages: []types.Package{setUpPkg(types.BundleChartVariable{
+						Name:    "VAR1",
+						Path:    "a.complex.path",
+						Default: "set-by-bundle",
+					})},
+				},
+			},
+			expectedKey: "VAR1",
+			expectedVal: "set-by-bundle",
+		},
+		{
+			name: "mask env var",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{},
+				},
+				bundle: types.UDSBundle{
+					Packages: []types.Package{setUpPkg(types.BundleChartVariable{Name: "VAR1", Path: "path"})},
+				},
+			},
+			loadEnv:     true,
+			envKey:      "UDS_VAR1",
+			envVal:      "gets-masked",
+			expectedKey: "VAR1",
+			expectedVal: "****",
+		},
+		{
+			name: "mask file var",
+			bundle: Bundle{
+				cfg: &types.BundleConfig{
+					DeployOpts: types.BundleDeployOptions{
+						Config: "uds-config.yaml",
+						Variables: map[string]map[string]interface{}{
+							pkgName: {
+								"VAR1": "../../test/bundles/07-helm-overrides/variable-files/test.cert",
+							},
+						},
+					},
+				},
+				bundle: types.UDSBundle{
+					Packages: []types.Package{setUpPkg(types.BundleChartVariable{
+						Name: "VAR1",
+						Path: "path",
+						Type: "file",
+					})},
+				},
+			},
+			expectedKey: "VAR1",
+			expectedVal: "****",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.loadEnv {
+				os.Setenv(tc.envKey, tc.envVal)
+			}
+
+			pkgViews := formPkgViews(&tc.bundle)
+
+			v := pkgViews[0].overrides["overrides"].([]interface{})[0].(map[string]map[string]interface{})[chartName]["variables"]
+
+			require.Contains(t, v.(map[string]interface{})[tc.expectedKey], tc.expectedVal)
+
+			// cleanup ENV Vars to avoid conflicts
+			if tc.loadEnv {
+				os.Unsetenv(tc.envKey)
+			}
+		})
+	}
+
+	zarfVarTest := TestCase{
```

Comment:
I can't seem to run this test in my editor. Is it possible to combine all the test cases into a single list of `[]TestCase`?